packages/server/lib/video_capture.js

```diff
@@ -217,62 +256,93 @@ module.exports = {
     })
   },
 
-  process (name, cname, videoCompression, onProgress = function () {}) {
-    let total = null
+  process (name, cname, videoCompression, ffmpegchaptersConfig, onProgress = function () {}) {
+    const metaFileName = `${name}.meta`
 
-    return new Promise((resolve, reject) => {
-      debug('processing video from %s to %s video compression %o',
-        name, cname, videoCompression)
-
-      ffmpeg()
-      .input(name)
-      .videoCodec('libx264')
-      .outputOptions([
-        '-preset fast',
-        `-crf ${videoCompression}`,
-      ])
-      // .videoFilters("crop='floor(in_w/2)*2:floor(in_h/2)*2'")
-      .on('start', (command) => {
-        return debug('compression started %o', { command })
-      }).on('codecData', (data) => {
-        debug('compression codec data: %o', data)
-
-        total = utils.timemarkToSeconds(data.duration)
-      }).on('stderr', (stderr) => {
-        return debug('compression stderr log %o', { message: stderr })
-      }).on('progress', (progress) => {
-        // bail if we dont have total yet
-        if (!total) {
-          return
-        }
+    const maybeGenerateMetaFile = Promise.method(() => {
+      if (!ffmpegchaptersConfig) {
+        return false
+      }
 
-        debug('compression progress: %o', progress)
+      // Writing the metadata to filesystem is necessary because fluent-ffmpeg is just a wrapper of ffmpeg command.
+      return fs.writeFile(metaFileName, ffmpegchaptersConfig).then(() => true)
+    })
+
+    return maybeGenerateMetaFile()
+    .then((addChaptersMeta) => {
```

Comment:
you could also use `async`...`await` in `videoCapture.process` instead of using this old promise-style syntax, it would minimize the diff

Reply:
Changed it.